tests/conftest.py

```diff
@@ -410,10 +413,22 @@
 ]
 
 # ports for testing, call next() on to avoid assignment overlap
-DEFAULT_HOST = "127.0.0.1"
+DEFAULT_HOST = LOCAL_HOST.hostname
 default_ports = itertools.count(10234)
 
 
+def remove_test_directory(directory: str) -> None:
+    """Destroy a directory once tests are done, change permissions if needed"""
+
+    def readonly_handler(func, path, execinfo) -> None:
+        """If permission is readonly, we change these and retry."""
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    # we need `onerror` to deal with permissions, e.g. on Windows
+    shutil.rmtree(directory, onerror=readonly_handler)
```

Comment:
This is a conftest file and it would make more sense to create a fixture that sets up a tmp folder, yields it, and then removes the test directory.

`BaseP2PLibp2pTest` can be entirely removed with fixtures.

Reply:
yes, but I want this utility function to be able to be used by all other places (primarily tests) where we need to delete a directory recursively during clean up